implementation/leprechaun/src/config/config.ts

```diff
@@ -0,0 +1,15 @@
+interface AppConfig {
+  PROD: boolean;
+  LEPRECHAUN_APP_PORT: number;
+  LEPRECHAUN_DATABASE_HOST: string;
+  LEPRECHAUN_DATABASE_DBNAME: string;
+}
+
+const config: AppConfig = {
+  PROD: process.env.NODE_ENV === 'production',
+  LEPRECHAUN_APP_PORT: parseInt(process.env.LEPRECHAUN_APP_PORT!) ?? 3000,
+  LEPRECHAUN_DATABASE_HOST: process.env.LEPRECHAUN_DATABASE_HOST! ?? 'postgresql://postgres:test@localhost:5432',
+  LEPRECHAUN_DATABASE_DBNAME: process.env.LEPRECHAUN_DATABASE_HOST! ?? 'leprechaun',
```

Comment:
On the other hand `??` is perfect here, but I don't know why you'd need `!` before them.

Reply:
Without the `!` in `parseInt` TS would complain about undefined not being assignable to `string`, and also, before adding `??` i just had `<name>: <env>` and it would also complain about undefined not being assignable to whatever type `AppConfig` needed.